diff --git a/dist/cjs/index.js b/dist/cjs/index.js
index 221b21a34f9681afd54840a5d8f725fe69b890a3..16cdbf6502c4d5f969442c6e410a2df797b8def7 100644
--- a/dist/cjs/index.js
+++ b/dist/cjs/index.js
@@ -31,7 +31,7 @@ var r = String.raw;
 var envFlags = {
   flagGroups: (() => {
     try {
-      new RegExp("(?i:)");
+      Reflect.construct(RegExp, ["(?i:)"]);
     } catch {
       return false;
     }
@@ -39,7 +39,7 @@ var envFlags = {
   })(),
   unicodeSets: (() => {
     try {
-      new RegExp("[[]]", "v");
+      Reflect.construct(RegExp, ["[[]]", "v"]);
     } catch {
       return false;
     }
@@ -48,7 +48,7 @@ var envFlags = {
 };
 envFlags.bugFlagVLiteralHyphenIsRange = envFlags.unicodeSets ? (() => {
   try {
-    new RegExp(r`[\d\-a]`, "v");
+    Reflect.construct(RegExp, [r`[\d\-a]`, "v"]);
   } catch {
     return true;
   }
diff --git a/dist/esm/index.js b/dist/esm/index.js
index eb64720c4918d4afeece67ccd8fd3c1b8e4997ce..b898d4d9726b9cb1bfccac45646f17789ea37636 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -4,7 +4,7 @@ var r = String.raw;
 var envFlags = {
   flagGroups: (() => {
     try {
-      new RegExp("(?i:)");
+      Reflect.construct(RegExp, ["(?i:)"]);
     } catch {
       return false;
     }
@@ -12,7 +12,7 @@ var envFlags = {
   })(),
   unicodeSets: (() => {
     try {
-      new RegExp("[[]]", "v");
+      Reflect.construct(RegExp, ["[[]]", "v"]);
     } catch {
       return false;
     }
@@ -21,7 +21,7 @@ var envFlags = {
 };
 envFlags.bugFlagVLiteralHyphenIsRange = envFlags.unicodeSets ? (() => {
   try {
-    new RegExp(r`[\d\-a]`, "v");
+    Reflect.construct(RegExp, [r`[\d\-a]`, "v"]);
   } catch {
     return true;
   }
